src/components/pages/SingleArt/SingleArt.tsx

```diff
@@ -0,0 +1,92 @@
+import { useParams, useNavigate } from "react-router";
+import galleryData from "../Gallery/galleryData";
+import NotFound from "../NotFound/NotFound";
+import ImageModal from "../../features/ImageModal/ImageModal";
+import Button from "../../common/Button/Button";
+import RedirectLink from "../../common/RedirectLink/RedirectLink";
+import styles from "./SingleArt.module.scss";
+import { useState } from "react";
+import { GrFormPreviousLink } from "react-icons/gr";
+import { GrFormNextLink } from "react-icons/gr";
+import useScroll from "../../../hooks/useScroll";
+import useTitle from "../../../hooks/useTitle";
+
+const SingleArt: React.FC = () => {
+  const navigate = useNavigate();
+  const [modal, setModal] = useState(false);
+  const { id } = useParams<{ id: string }>();
```

Comment:
Skoro to boolean, to warto byłoby zmienić tę nazwę. Np. na `showModal`.

Dodatkowo w stanie lokalnym możemy też precyzyjnie określać typ.

```
useState<Boolean>(false);
```

...i warto to robić :)